cmd/root/ssh/ssh.go

```diff
@@ -50,16 +53,62 @@ func (c *CmdSSHActions) CopyFromHost(hostName string, _ bool, source, destinatio
 	return c.cli.Execute("scp", "-i", c.sshKey, "-o", "UserKnownHostsFile=/dev/null", "-o", "StrictHostKeyChecking=no", fmt.Sprintf("%v@%v:%v", c.sshUser, hostName, source), destination)
 }
 
-func (c *CmdSSHActions) CopyFromHostSudo(hostName string, _ bool, _, source, destination string) (string, error) {
-	return c.cli.Execute("scp", "-i", c.sshKey, "-o", "UserKnownHostsFile=/dev/null", "-o", "StrictHostKeyChecking=no", fmt.Sprintf("%v@%v:%v", c.sshUser, hostName, source), destination)
+func (c *CmdSSHActions) CopyFromHostSudo(hostName string, _ bool, sudoUser, source, destination string) (string, error) {
+	sourceFileName := filepath.Base(source)
+	// create a tmp dir for scp
+	tmpDir := path.Join("/tmp/", "ddc-scp-"+uuid.New().String())
+	_, err := c.HostExecute(hostName, false, "mkdir", "-p", tmpDir)
+	if err != nil {
+		return "", err
+	}
+	// set perms
+	_, err = c.HostExecute(hostName, false, "chmod", "a+rwx", tmpDir)
+	if err != nil {
+		return "", err
+	}
+	// cleanup the tmp dir
+	defer func() {
+		c.HostExecute(hostName, false, "rm", "-rf", tmpDir)
+	}()
+	// first move to tmp dir from source as sudo
+	tmpFilePath := path.Join(tmpDir, sourceFileName)
+	_, err = c.HostExecuteSudo(hostName, sudoUser, "cp", source, tmpFilePath)
+	if err != nil {
+		return "", err
+	}
+	// next copy from tmp dir as non-sudo
+	return c.cli.Execute("scp", "-i", c.sshKey, "-o", "UserKnownHostsFile=/dev/null", "-o", "StrictHostKeyChecking=no", tmpFilePath, fmt.Sprintf("%v@%v:%v", c.sshUser, hostName, destination))
 }
 
 func (c *CmdSSHActions) CopyToHost(hostName string, _ bool, source, destination string) (string, error) {
 	return c.cli.Execute("scp", "-i", c.sshKey, "-o", "UserKnownHostsFile=/dev/null", "-o", "StrictHostKeyChecking=no", source, fmt.Sprintf("%v@%v:%v", c.sshUser, hostName, destination))
 }
 
-func (c *CmdSSHActions) CopyToHostSudo(hostName string, _ bool, _, source, destination string) (string, error) {
-	return c.cli.Execute("scp", "-i", c.sshKey, "-o", "UserKnownHostsFile=/dev/null", "-o", "StrictHostKeyChecking=no", source, fmt.Sprintf("%v@%v:%v", c.sshUser, hostName, destination))
+func (c *CmdSSHActions) CopyToHostSudo(hostName string, _ bool, sudoUser, source, destination string) (string, error) {
+	sourceFileName := filepath.Base(source)
+	// create a tmp dir for scp
+	tmpDir := path.Join("/tmp/", "ddc-scp-"+uuid.New().String())
+	_, err := c.HostExecute(hostName, false, "mkdir", "-p", tmpDir)
+	if err != nil {
+		return "", err
+	}
+	// set perms
+	_, err = c.HostExecute(hostName, false, "chmod", "a+rwx", tmpDir)
+	if err != nil {
+		return "", err
+	}
+	// cleanup the tmp dir
+	defer func() {
+		c.HostExecute(hostName, false, "rm", "-rf", tmpDir)
```

Comment:
log an error here (debug level)

also the defer should be after the create not after the chmod